src/main/java/org/ethereum/beacon/discovery/util/Functions.java

```diff
@@ -15,153 +12,154 @@
 import java.math.BigInteger;
 import java.nio.ByteOrder;
 import java.security.SecureRandom;
+import java.security.Security;
 import java.util.Random;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.crypto.Hash;
+import org.apache.tuweni.crypto.SECP256K1;
+import org.apache.tuweni.crypto.SECP256K1.KeyPair;
+import org.apache.tuweni.crypto.SECP256K1.Parameters;
+import org.apache.tuweni.crypto.SECP256K1.PublicKey;
+import org.apache.tuweni.crypto.SECP256K1.SecretKey;
+import org.apache.tuweni.crypto.SECP256K1.Signature;
 import org.bouncycastle.crypto.Digest;
 import org.bouncycastle.crypto.digests.SHA256Digest;
 import org.bouncycastle.crypto.generators.HKDFBytesGenerator;
-import org.bouncycastle.crypto.params.ECDomainParameters;
 import org.bouncycastle.crypto.params.HKDFParameters;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.bouncycastle.math.ec.ECPoint;
-import org.bouncycastle.util.Arrays;
 import org.ethereum.beacon.discovery.type.Hashes;
-import org.web3j.crypto.ECDSASignature;
-import org.web3j.crypto.ECKeyPair;
-import org.web3j.crypto.Hash;
-import org.web3j.crypto.Sign;
 
 /** Set of cryptography and utilities functions used in discovery */
 public class Functions {
   private static final Logger logger = LogManager.getLogger();
-  public static final ECDomainParameters SECP256K1_CURVE =
-      new ECDomainParameters(
-          CURVE_PARAMS.getCurve(), CURVE_PARAMS.getG(), CURVE_PARAMS.getN(), CURVE_PARAMS.getH());
+  private static final BouncyCastleProvider PROVIDER;
   public static final int PRIVKEY_SIZE = 32;
   public static final int PUBKEY_SIZE = 64;
+  public static final int COMPRESSED_PUBKEY_SIZE = 33;
   private static final int RECIPIENT_KEY_LENGTH = 16;
   private static final int INITIATOR_KEY_LENGTH = 16;
   private static final int AUTH_RESP_KEY_LENGTH = 16;
   private static final int MS_IN_SECOND = 1000;
 
   private static final Supplier<SecureRandom> SECURE_RANDOM = Suppliers.memoize(SecureRandom::new);
 
-  private static boolean SKIP_SIGNATURE_VERIFY = false;
+  static {
+    Security.addProvider(PROVIDER = new BouncyCastleProvider());
+  }
 
   /** SHA2 (SHA256) */
-  public static Bytes hash(Bytes value) {
-    return Hashes.sha256(value);
+  public static Bytes32 hash(final Bytes value) {
+    return Bytes32.wrap(Hashes.sha256(value));
   }
 
   /** SHA3 (Keccak256) */
-  public static Bytes hashKeccak(Bytes value) {
-    return Bytes.wrap(Hash.sha3(value.toArray()));
+  public static Bytes32 hashKeccak(final Bytes value) {
+    return Bytes32.wrap(Hash.keccak256(value.toArray()));
```

Comment:
Wouldn't the below be enough?

```suggestion
    return Hash.keccak256(value);
```